log4j-core/src/main/java/org/apache/logging/log4j/core/async/DisruptorUtil.java

```diff
@@ -60,36 +60,44 @@
     private DisruptorUtil() {
     }
 
-    static long getTimeout(final String propertyName, final long defaultTimeout) {
-        return PropertiesUtil.getProperties().getLongProperty(propertyName, defaultTimeout);
+    static WaitStrategy createWaitStrategy(final String propertyName) {
+        final String strategyStr = PropertiesUtil.getProperties().getStringProperty(propertyName, DisruptorWaitStrategy.TIMEOUT.toString());
+        LOGGER.trace("property {}={}", propertyName, strategyStr);
+        final DisruptorWaitStrategy strategy = DisruptorWaitStrategy.valueOf(Strings.toRootUpperCase(strategyStr));
+        final long timeoutMillis = parseAdditionalLongProperty(propertyName, "Timeout", 10L);
+        switch (strategy) {
+            case SLEEP:
+                final long sleepTimeNs =
+                        parseAdditionalLongProperty(propertyName, "SleepTimeNs", 100L);
+                final String key = getFullPropertyKey(propertyName, "Retries");
+                final int retries =
+                        PropertiesUtil.getProperties().getIntegerProperty(key, 200);
+                return new SleepingWaitStrategy(retries, sleepTimeNs);
+            case YIELD:
+                return new YieldingWaitStrategy();
+            case BLOCK:
+                return new BlockingWaitStrategy();
+            case BUSYSPIN:
+                return new BusySpinWaitStrategy();
+            case TIMEOUT:
+                return new TimeoutBlockingWaitStrategy(timeoutMillis, TimeUnit.MILLISECONDS);
+            default:
+                return new TimeoutBlockingWaitStrategy(timeoutMillis, TimeUnit.MILLISECONDS);
+        }
     }
 
-    static WaitStrategy createWaitStrategy(final String propertyName) {
-        final String key = propertyName.startsWith("AsyncLogger.")
-                ? "AsyncLogger.Timeout"
-                : "AsyncLoggerConfig.Timeout";
-        final long timeoutMillis = DisruptorUtil.getTimeout(key, 10L);
-        return createWaitStrategy(propertyName, timeoutMillis);
+    private static String getFullPropertyKey(final String strategyKey, final String additionalKey) {
+        return strategyKey.startsWith("AsyncLogger.")
+                ? "AsyncLogger." + additionalKey
+                : "AsyncLoggerConfig." + additionalKey;
     }
 
-    static WaitStrategy createWaitStrategy(final String propertyName, final long timeoutMillis) {
-        final String strategy = PropertiesUtil.getProperties().getStringProperty(propertyName, "TIMEOUT");
-        LOGGER.trace("property {}={}", propertyName, strategy);
-        final String strategyUp = strategy.toUpperCase(Locale.ROOT); // TODO Refactor into Strings.toRootUpperCase(String)
-        switch (strategyUp) { // TODO Define a DisruptorWaitStrategy enum?
```

Comment:
This TODO comment ("Define an enum") is misleading. Let's replace this with a comment explaining that we deliberately don't use an enum so that invalid values will result in the default timeoutstrategy being chosen without exceptions being thrown.

(As an aside, I believe the value of defining a private enum is limited anyway. What matters to me is that the valid values are well-documented and that this documentation is easy to find.)

Reply:
Ok. TODO removed. Proper comment on deliberate usage is added.

Regrding documentation, I haven`t found any documentation on property related to `timeoutMillis` parameter of `TimeoutBlockingWaitStrategy`. I think the best place is to add documentation on all these three properties (`timeoutMillis`, `sleepTimeNs`, `retries`) to `logging-log4j2/src/site/asciidoc/manual/async.adoc`. Do you think it is worth to do it? And what is the best place then?